server/athenian/api/controllers/selector.py

```diff
@@ -0,0 +1,80 @@
+from typing import Callable, Optional
+
+import aiomcache
+from sqlalchemy import and_, select
+
+from athenian.api.controllers.features.entries import get_calculator
+from athenian.api.models.state.models import AccountFeature, Feature, \
+    FeatureComponent, God
+from athenian.api.typing_utils import DatabaseLike
+
+
+async def get_calculator_for_user(
+    service: str,
+    account_id: int,
+    user_id: str,
+    sdb: DatabaseLike,
+    mdb: DatabaseLike,
+    pdb: DatabaseLike,
+    rdb: DatabaseLike,
+    cache: Optional[aiomcache.Client],
+    raise_err: Optional[bool] = False,
+    base_module: Optional[str] = "athenian.api.experiments",
+) -> Callable:
+    """Get the metrics calculator function for the given user."""
+
+    def _get_calculator(variation=None):
+        return get_calculator(
+            service, mdb, pdb, rdb, cache, variation=variation, raise_err=raise_err,
+            base_module=base_module,
+        )
+
+    feature_name_prefix = METRIC_ENTRIES_VARIATIONS_PREFIX[service]
+    all_metrics_variations_features = await sdb.fetch_all(
+        select([Feature.id, Feature.name, Feature.default_parameters]).where(
+            and_(
+                Feature.name.like(f"{feature_name_prefix}%"),
+                Feature.component == FeatureComponent.server,
+                Feature.enabled,
+            ),
+        ),
+    )
+
+    if not all_metrics_variations_features:
+        return _get_calculator()
+
+    all_metrics_variations_features = {
+        row[0]: {"name": row[1][len(feature_name_prefix):], "params": row[2] or {}}
+        for row in all_metrics_variations_features
+    }
+
+    metrics_variation_feature = await sdb.fetch_one(
+        select([AccountFeature.feature_id, AccountFeature.parameters]).where(
+            and_(
+                AccountFeature.account_id == account_id,
+                AccountFeature.feature_id.in_(all_metrics_variations_features),
+                AccountFeature.enabled,
+            ),
+        ),
+    )
+
+    if not metrics_variation_feature:
+        return _get_calculator()
+
+    selected_metrics_variation = all_metrics_variations_features[
+        metrics_variation_feature[0]
+    ]
+    metrics_variation_params = {
+        **selected_metrics_variation["params"],
+        **(metrics_variation_feature[1] or {}),
+    }
+
+    is_god = await sdb.fetch_one(select([God.user_id]).where(God.user_id == user_id))
```

Comment:
We have `request.god_id` defined in god mode.

If that does not fit, we should batch this and one of the previous queries together in `gather()`. Otherwise, we've got 3 queries to sdb standing on the critical path, that's too much.

Is it possible to merge the previous two queries together somehow? A join maybe?

Reply:
`request.god_id` will work, thanks for pointing it out, and I agree with the perf bottleneck here otherwise 👍